types/block.go

```diff
@@ -100,9 +100,10 @@ func (dah *DataAvailabilityHeader) Equals(to *DataAvailabilityHeader) bool {
 
 // Hash computes and caches the merkle root of the row and column roots.
 func (dah *DataAvailabilityHeader) Hash() []byte {
-	if dah == nil {
-		return merkle.HashFromByteSlices(nil)
+	if dah.IsZero() {
+		dah = MinDataAvailabilityHeader()
 	}
```

Comment:
@evan-forbes, I guess this is wrong. Zero DAH is when rows len and cols len are equal to zero, while MinDataAvailabilityHeader has 2 rows and 2 cols, as empty block data does not mean not data at all

Reply:
I would just remove this if

Reply:
Or the intuition here is to never actually use empty DAH and always swap it with empty DAH, right?